pkg/autoscaler/metrics/collector.go

```diff
@@ -385,6 +370,41 @@ func (c *collection) record(stat Stat) {
 	c.rpsPanicBuckets.Record(stat.Time, rps)
 }
 
+// startScraping starts scraping metrics.
+// This is only run from inside a lock, it can't race with a stopScraping call.
+func (c *collection) startScraping() {
+	c.stopCh = make(chan struct{})
+	c.grp.Add(1)
+	go func() {
+		defer c.grp.Done()
+
+		scrapeTicker := c.tickFactory(scrapeTickInterval)
+		defer scrapeTicker.Stop()
+		for {
+			select {
+			case <-c.stopCh:
+				return
+			case <-scrapeTicker.C:
+				metric := c.currentMetric()
+				stat, err := c.getScraper().Scrape(metric.Spec.StableWindow)
+				if c.updateLastError(err) {
+					c.callback(types.NamespacedName{Namespace: metric.Namespace, Name: metric.Name})
```

Comment:
this will be constructed every second, while it's constant. probably we can just preallocate one outside of the loop.